src/main/java/net/minecraftforge/common/DimensionManager.java

```diff
@@ -286,39 +317,65 @@ public static WorldProvider createProviderFor(int dim)
         }
     }
 
-    public static void unloadWorld(int id) {
-        unloadQueue.add(id);
+    /**
+     * Queues a dimension to unload.
+     * If the dimension is already queued, it will reset the delay to unload
+     * @param id The id of the dimension
+     */
+    public static void unloadWorld(int id)
+    {
+        if(!unloadQueue.contains(id))
+        {
+            FMLLog.fine("Queueing dimension %s to unload", id);
+            unloadQueue.add(id);
+        }
+        else
+        {
+            dimensions.get(id).ticksWaited = 0;
+        }
+    }
+
+    public static boolean isWorldQueuedToUnload(int id)
+    {
+        return unloadQueue.contains(id);
     }
 
     /*
     * To be called by the server at the appropriate time, do not call from mod code.
     */
     public static void unloadWorlds(Hashtable<Integer, long[]> worldTickTimes) {
+        IntArrayList toRemove = new IntArrayList();
         for (int id : unloadQueue) {
```

Comment:
Instead of creating `toRemove`, can you use an iterator on `unloadQueue` and use `iterator.remove()`?